### PR TITLE
adds tests for tablet adding new file to metadata

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      // update the tablets time, so it will differ
+      testAmple.mutateTablet(extent).putTime(tabletTime.getMetadataTime(tabletTime.getTime() + 2))
+          .mutate();
+
+      tabletTime.updateTimeIfGreater(tabletTime.getTime() + 6);
+
+      updateTabletDataFile(testAmple, tabletTime.getTime(), newFile, dfv1, Set.of(), 7L,
+          MinorCompactionReason.SYSTEM, tserverInstance, extent, lastMetadata, tabletTime,
+          flushNonce);
+
```
Comment (dlmarion) [inline test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java:114]: Is there a case where the tablet's time has changed in the metadata table, maybe from some other operation performed in the Manager? This looks like it tests the positive case, is there a negative case?

File: test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      assertNull(testAmple.readTablet(extent));
+
+      // create tablet in the fake metadata table
+      testAmple.mutateTablet(extent).putDirName("dir1").putTime(tabletTime.getMetadataTime())
+          .putLocation(Location.current(tserverInstance)).putPrevEndRow(null).mutate();
+
+      // get the tablets metadata
+      var lastMetadata = testAmple.readTablet(extent);
+
+      // TODO open an issue about making requireLocation more strict
```
Comment (dlmarion) [inline test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java:153]: Likely need to resolve this before merging

Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java]: Removed the todo and opened #4620.  While working on this I wrote a buggy version of this test that did not fail in the way I expected and discovered #4620 in the process.  After discovering the test bug I fixed the test and added the todo.

File: test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java (hunk 1 above)
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java:114]: Yeah there is a good negative case to test.  Added that in 74b8f14

**Review by dlmarion — APPROVED** [review]

File: test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    var tabletTime = TabletTime.getInstance(new MetadataTime(1, TimeType.LOGICAL));
+    long flushNonce = 42L;
+
+    try (var client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(metadataTable);
+      TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      // create test ample that will return unknown status for conditional mutations.
+      Ample testAmple = TestAmple.create(getCluster().getServerContext(),
```
Comment (cshannon) [inline test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java:243]: It's nice that we can easily test the rejection handlers with different simulated states now that we have `TestAmple`, hopefully we can go back and add some more tests like this for other rejection handlers over time.

**Pull request merged**